### PR TITLE
Code Cleanup

### PR DESCRIPTION
**Pull request by mustafa-cosar**

### CODE REVIEW

File: src/int64_util.js [hunk 1]
```diff
@@ -17,85 +17,112 @@
...
+
+var BYTE_7 = 7;
+var BYTE_6 = 6;
+var BYTE_5 = 5;
+var BYTE_4 = 4;
+var BYTE_3 = 3;
+var BYTE_2 = 2;
+var BYTE_1 = 1;
+
+var MAX_NUMBER_OF_DECIMAL_DIGITS_REPRESENTABLE_BY_DOUBLE = 15;
```
Comment (thehcma) [inline src/int64_util.js:37]: MAX_NUM...

File: src/int64_util.js [hunk 2]
```diff
@@ -17,85 +17,112 @@
...
+var NUMBER_OF_LEAST_SIGNIFICANT_DIGITS = 11;
+
 var POW2_24 = Math.pow(2, 24);
```
Comment (thehcma) [inline src/int64_util.js:40]: sort

File: src/int64_util.js [hunk 3]
```diff
@@ -17,85 +17,112 @@
...
+var BYTE_7 = 7;
+var BYTE_6 = 6;
+var BYTE_5 = 5;
+var BYTE_4 = 4;
+var BYTE_3 = 3;
+var BYTE_2 = 2;
+var BYTE_1 = 1;
+
+var MAX_NUMBER_OF_DECIMAL_DIGITS_REPRESENTABLE_BY_DOUBLE = 15;
+var NUMBER_OF_LEAST_SIGNIFICANT_DIGITS = 11;
```
Comment (thehcma) [inline src/int64_util.js:38]: NUM_...

File: src/int64_util.js [hunk 4]
```diff
@@ -17,85 +17,112 @@
...
-var POW10_11 = Math.pow(10, 11);
+var POW2_48 = Math.pow(2, 48);
+var POW10_11 = Math.pow(10, NUMBER_OF_LEAST_SIGNIFICANT_DIGITS);
+var POW10_15 = Math.pow(10, MAX_NUMBER_OF_DECIMAL_DIGITS_REPRESENTABLE_BY_DOUBLE);
+var POW2_48_OVER_POW10_11 = Math.floor(POW2_48 / POW10_11);
+var POW2_48_REMAINDER_POW10_11 = Math.floor(POW2_48 % POW10_11);
+var POW10_15_OVER_POW2_32 = Math.floor(POW10_15 / POW2_32);
+var POW10_15_REMAINDER_POW2_32 = Math.floor(POW10_15 % POW2_32);
+
+var FOUR_BYTES_BITMASK = 0xffffffff;
```
Comment (thehcma) [inline src/int64_util.js:51]: FOUR_BYTE (both)

File: src/int64_util.js [hunk 5]
```diff
@@ -17,85 +17,112 @@
...
+var FOUR_BYTES_HIGH_1_BITMASK = 0x80000000;
+var ONE_BYTE_BITMASK = 0xff;
+var ONE_BYTE_HIGH_1_BITMASK = 0x80;
+var ONE_BYTE_HIGH_3_BITMASK = 0xe0;
+
+var ONE_BYTE_BIT_LENGTH = 8;
+var TWO_BYTES_BIT_LENGTH = 16;
+
+var MAX_NUMBER_OF_CHARACTERS_FOR_INT64 = 19;
```
Comment (thehcma) [inline src/int64_util.js:60]: MAX_NUM_....

File: src/int64_util.js [hunk 6]
```diff
@@ -17,85 +17,112 @@
...
-    low = ('00000000000' + String(low % POW10_11)).slice(-11);
+    low = ('00000000000' + String(low % POW10_11)).slice(-NUMBER_OF_LEAST_SIGNIFICANT_DIGITS);
     return (negative ? '-' : '') + String(high) + low;
 };
 
 Int64Util.fromDecimalString = function (text)
 {
     var negative = text.charAt(0) === '-';
-    if (text.length < (negative ? 17 : 16))
+    if (text.length < (negative ? MIN_NUMBER_OF_CHARACTERS_FOR_INT64 + 1 : MIN_NUMBER_OF_CHARACTERS_FOR_INT64))
```
Comment (thehcma) [inline src/int64_util.js:109]: MIN_NUM...

File: src/parse.js [hunk 7]
```diff
@@ -30,177 +29,269 @@
...
+                    if (ch === ']')
+                    {
+                        next(']');
+                        return array;
+                    }
+                    next(',');
+                    white();
+                }
+            }
+            error("Bad array");
```
Comment (thehcma) [inline src/parse.js:93]: lower case first char

File: src/parse.js [hunk 8]
```diff
@@ -30,177 +29,269 @@
...
+        next = function (c)
+        {
+            // If a c parameter is provided, verify that it matches the current character.
 
-// If a c parameter is provided, verify that it matches the current character.
-
-            if (c && c !== ch) {
+            if (c && c !== ch)
+            {
                 error("Expected '" + c + "' instead of '" + ch + "'");
```
Comment (thehcma) [inline src/parse.js:114]: first char in lowercase everywhere

File: src/parse.js [hunk 9]
```diff
@@ -30,177 +29,269 @@
...
+                    if (ch === '}')
+                    {
+                        next('}');
+                        return object;
+                    }
+                    next(',');
+                    white();
+                }
+            }
+            error("Bad object");
```
Comment (thehcma) [inline src/parse.js:219]: lowercase

File: src/parse.js [hunk 10]
```diff
@@ -222,78 +313,15 @@ var json_parse = function (options) {
                 return null;
             }
             error("Unexpected '" + ch + "'");
```
Comment (thehcma) [inline src/parse.js:315]: lowercase

File: src/int64_util.js [hunk 11]
```diff
@@ -17,85 +17,112 @@
...
+var ONE_BYTE_HIGH_1_BITMASK = 0x80;
+var ONE_BYTE_HIGH_3_BITMASK = 0xe0;
+
+var ONE_BYTE_BIT_LENGTH = 8;
+var TWO_BYTES_BIT_LENGTH = 16;
+
+var MAX_NUMBER_OF_CHARACTERS_FOR_INT64 = 19;
+var MIN_NUMBER_OF_CHARACTERS_FOR_INT64 = 16;
 
 Int64Util.toDecimalString = function (i64)
```
Comment (thehcma) [inline src/int64_util.js:63]: do we have unit tests for all these functions?

**Review by thehcma — APPROVED** [review]

File: src/int64_util.js (hunk 1 above)
Comment (mustafa-cosar) [reply src/int64_util.js:37]: fixed

File: src/int64_util.js (hunk 2 above)
Comment (mustafa-cosar) [reply src/int64_util.js:40]: sorted

File: src/int64_util.js (hunk 3 above)
Comment (mustafa-cosar) [reply src/int64_util.js:38]: fixed

File: src/int64_util.js (hunk 4 above)
Comment (mustafa-cosar) [reply src/int64_util.js:51]: fixed

File: src/int64_util.js (hunk 5 above)
Comment (mustafa-cosar) [reply src/int64_util.js:60]: fixed

File: src/int64_util.js (hunk 6 above)
Comment (mustafa-cosar) [reply src/int64_util.js:109]: fixed

File: src/parse.js (hunk 7 above)
Comment (mustafa-cosar) [reply src/parse.js:93]: fixed

File: src/parse.js (hunk 8 above)
Comment (mustafa-cosar) [reply src/parse.js:114]: fixed

File: src/int64_util.js (hunk 11 above)
Comment (mustafa-cosar) [reply src/int64_util.js:63]: yes, we do

File: src/parse.js (hunk 10 above)
Comment (mustafa-cosar) [reply src/parse.js:315]: fixed

File: src/parse.js (hunk 9 above)
Comment (mustafa-cosar) [reply src/parse.js:219]: fixed

**Pull request merged**